stdlib/experimental/chain.go

```diff
@@ -0,0 +1,70 @@
+package experimental
+
+import (
+	"context"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/interpreter"
+	"github.com/influxdata/flux/lang"
+	"github.com/influxdata/flux/lang/execdeps"
+	"github.com/influxdata/flux/runtime"
+	"github.com/influxdata/flux/values"
+)
+
+func init() {
+	runtime.RegisterPackageValue("experimental", "chain", MakeChainFunction())
+}
+
+func MakeChainFunction() values.Function {
+	chainSignature := runtime.MustLookupBuiltinType("experimental", "chain")
+	return values.NewFunction("chain", chainSignature, chainCall, false)
+}
+
+func chainCall(ctx context.Context, args values.Object) (values.Value, error) {
+	arguments := interpreter.NewArguments(args)
+
+	first, err := arguments.GetRequired("first")
+	if err != nil {
+		return nil, err
+	}
+
+	second, err := arguments.GetRequired("second")
+	if err != nil {
+		return nil, err
+	}
+
+	compiler := lang.TableObjectCompiler{
+		Tables: first.(*flux.TableObject),
+	}
+
+	program, err := compiler.Compile(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, codes.Inherit, "error in table object compilation")
+	}
+
+	if !execdeps.HaveExecutionDependencies(ctx) {
+		return nil, errors.New(codes.Internal, "no execution context for chain to use")
+	}
+	deps := execdeps.GetExecutionDependencies(ctx)
+
+	if program, ok := program.(lang.LoggingProgram); ok {
+		program.SetLogger(deps.Logger)
+	}
+	query, err := program.Start(ctx, deps.Allocator)
+	if err != nil {
+		return nil, errors.Wrap(err, codes.Inherit, "error in table object start")
+	}
+
+	for res := range query.Results() {
+		if err := res.Tables().Do(func(table flux.Table) error {
+			defer table.Done()
+			return nil
+		}); err != nil {
+			return nil, err
+		}
+	}
+
+	return second, nil
+}
```

Comment:
Wondering if there is an easy way to verify that `first` ran before `second` ... Only way I can think of is to write a custom function that introduces some delay and then sets some state, ensuring that the second function blocks on its completion.

Reply:
I spent a bunch of time trying to inject some mocks into the context to try and do it, but must be missing something about the execution. I'm going to round up some help today to see if I can't add more assertions to it.